pkg/cmd/server/server.go

```diff
@@ -291,13 +305,50 @@ func (s *server) run() error {
 		}
 	}
 
-	if err := s.runControllers(config, backupStorageLocation); err != nil {
+	if err := s.runControllers(config, backupStorageLocation, defaultVolumeSnapshotLocations); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func (s *server) getDefaultVolumeSnapshotLocations() (map[string]*api.VolumeSnapshotLocation, error) {
+	providerDefaults := make(map[string]*api.VolumeSnapshotLocation)
+	if len(s.config.defaultVolumeSnapshotLocations) == 0 {
+		return providerDefaults, nil
+	}
+
+	volumeSnapshotLocations, err := s.arkClient.ArkV1().VolumeSnapshotLocations(s.namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return providerDefaults, errors.WithStack(err)
+	}
+
+	providerLocations := make(map[string][]*api.VolumeSnapshotLocation)
+	for _, vsl := range volumeSnapshotLocations.Items {
+		providerLocations[vsl.Spec.Provider] = append(providerLocations[vsl.Spec.Provider], &vsl)
+	}
+
+	for provider, locations := range providerLocations {
+		defaultLocation, ok := s.config.defaultVolumeSnapshotLocations[provider]
+		if !ok {
+			return providerDefaults, errors.Errorf("missing provider %s. You must provide a default for every provider.", provider)
+		}
+
+		for _, location := range locations {
+			if location.ObjectMeta.Name == defaultLocation {
+				providerDefaults[provider] = location
+				break
+			}
+		}
+
+		if _, ok := providerDefaults[provider]; !ok {
+			return providerDefaults, errors.Errorf("missing provider %s. You must provide a default for every provider.", provider)
```

Comment:
this error should say that the default VSL specified for the provider isn't valid for that provider, and include both the provider & VSL name

Reply:
oops, good catch, copy pasta I forgot to update